### PR TITLE
[xgboost] fix nthread limiting via prop

### PR DESCRIPTION
**Pull request by honzasterba**

### CODE REVIEW

File: h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/XGBoostModelTest.java [hunk 1]
```diff
@@ -9,6 +9,14 @@
 
 public class XGBoostModelTest {
   
+  @Test
+  public void shouldGetCorrectMaxThread() {
+    assertEquals(Runtime.getRuntime().availableProcessors(), XGBoostModel.getMaxNThread());
+    System.setProperty("sys.ai.h2o.xgboost.nthreadMax", "2");
+    assertEquals(2, XGBoostModel.getMaxNThread());
+    System.clearProperty("sys.ai.h2o.xgboost.nthreadMax");
```
Comment (Pscheidl) [inline h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/XGBoostModelTest.java]: Proper, fault-tolerant cleanup required.

File: h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/XGBoostModelTest.java [hunk 2]
```diff
@@ -9,6 +9,14 @@
 
 public class XGBoostModelTest {
   
+  @Test
+  public void shouldGetCorrectMaxThread() {
+    assertEquals(Runtime.getRuntime().availableProcessors(), XGBoostModel.getMaxNThread());
+    System.setProperty("sys.ai.h2o.xgboost.nthreadMax", "2");
```
Comment (Pscheidl) [inline h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/XGBoostModelTest.java]: There is a dedicated JUnit library named [System rules](https://stefanbirkner.github.io/system-rules/) which will help you avoid all the problems associated with testing based on system properties. We already have the dependency (which you can update optionally) and it is already used in H2O.

If nothing else, it makes sure the environment variables are restored properly. See `SQLManagerTest` for example.

File: h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java [hunk 3]
```diff
@@ -35,7 +35,7 @@
         implements SharedTreeGraphConverter, Model.LeafNodeAssignment, Model.Contributions {
 
   private static final String PROP_VERBOSITY = H2O.OptArgs.SYSTEM_PROP_PREFIX + ".xgboost.verbosity";
-  private static final String PROP_NTHREAD = SYSTEM_PROP_PREFIX + "xgboost.nthread";
+  private static final String PROP_NTHREAD = SYSTEM_PROP_PREFIX + "xgboost.nthreadMax";
```
Comment (Pscheidl) [inline h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java]: This appears to require a change in documentation, yet there is no JIRA tied with this PR.

File: h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/XGBoostModelTest.java (hunk 2 above)
Comment (honzasterba) [reply h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/XGBoostModelTest.java]: thanks, used it

File: h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java (hunk 3 above)
Comment (honzasterba) [reply h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java]: the property is in experimental mode and has not been documented yet

**Pull request merged**